### PR TITLE
feat: log k8s create/update/delete actions

### PR DESCRIPTION
**Pull request by jopit**
**What type of PR is this?**

/kind enhancement

**What does this PR do / why we need it**:

Logs Create/Update/Deletion actions performed by Argo CD operator

### CODE REVIEW

**Review by jgwest — CHANGES_REQUESTED** [review]
Looks great, thanks @jopit! And thanks for going through all the code and adding all these statements, I appreciate it as it was likely both somewhat monotonous work :smile:, but also a case where its important to get the details right (and that combination can be deadly!)

I'm definitely looking forward to these changes making it into product code, as it should make working with user/customer operator logs much easier.

Everything looks great, just one suggestion around checking for `nil` on the logged object then good to go.

File: controllers/argoutil/resource.go [hunk 1]
```diff
@@ -122,3 +128,44 @@ func AnnotationsForCluster(cr *argoproj.ArgoCD) map[string]string {
...
+func LogResourceUpdate(log logr.Logger, object metav1.Object, explanations ...string) {
+	LogResourceAction(log, "Updating", object, explanations...)
+}
+
+func LogResourceDeletion(log logr.Logger, object metav1.Object, explanations ...string) {
+	LogResourceAction(log, "Deleting", object, explanations...)
+}
+
+func LogResourceAction(log logr.Logger, action string, object metav1.Object, explanations ...string) {
+	typeName := reflect.TypeOf(object).String()
```
[the remark this answers is not in the document]
Comment (jopit) [reply controllers/argoutil/resource.go:150]: Calling `log.Error(..)` already writes the stack trace to the log, so no need for the call to `debug.PrintStack()`

[the remark this answers is not in the document]
Comment (jopit) [reply controllers/argoutil/resource.go:150]: Done.

**Review by jgwest — APPROVED** [review]
LGTM, thanks again @jopit!

**Pull request merged**